### ISSUE TITLE
`got`: ssh not found

### ISSUE THREAD

**Issue by SZanko** [NixOS/nixpkgs#322043]
### Describe the bug
got clone doesn't work with ssh because it rely to find ssh in /usr/bin/ssh which doesn't work on nixos

### Steps To Reproduce
1. `nix-shell -p got`
2. `got clone git@github.com:github/docs.git`
3. `Connecting to ssh://git@github.com/github/docs.git
got: execv /usr/bin/ssh: No such file or directory
got-fetch-pack: unexpected end of file
got: unexpected end of file`

### Expected behavior
Clone the repository to disk

### Notify maintainers

@wahjava 
@afh 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.32, NixOS, 24.11 (Vicuña), 24.11.20240607.051f920`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/3dr5pyja36lvvrszhzffww1jwyrx6i09-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by afh** [comment]
Thanks for raising this, @SZanko, there is a conversation about this already going on in #297154. What's your take on this?

**Comment by FliegendeWurst** [comment]
Fixed with #322600
